### PR TITLE
Add most recent repos to all recent list

### PR DESCRIPTION
**Pull request by gerhardol**
Part of #4942 

## Proposed changes

Replace group of "less recent" repositories with "all recent" repositories list.
This simplifies my usage: Quick view at most recent that occasionally is to quick and fail, then a closer look at "all recent".
This can be seen as a step to deprecate "most recent" as suggested by RussKie (I really want to keep it though, even if few users know about it).

"Less recent" renamed to "Recent" in the UI, "All recent" in the code.
"All recent" is still default 0, you have to enable the functionality.
"Most recent" renamed to "Pinned" in the UI and the code.

### Before

image

### After

image

## Test methodology 

Manual

There are no tests for the form, just for the repo handling (that mentions "most recent" when handling the groups, not the same as "most recent repos").

## Merge strategy

First commit is the actual change, second is just cleaning up the use of "less recent repos".

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/UserRepositoryHistory/Repository.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class Repository
         public enum RepositoryAnchor
         {
             MostRecent,
```
Comment (RussKie) [inline GitCommands/UserRepositoryHistory/Repository.cs:12]: So would you say these are "pinned"?

**Review by RussKie — APPROVED** [review]
I still think we have a lot of unnecessary complexity here, but I'd take it as a step in the right direction :)

File: GitCommands/UserRepositoryHistory/Repository.cs (hunk 1 above)
Comment (pmiossec) [reply GitCommands/UserRepositoryHistory/Repository.cs:12]: +100
I was about to ask for renaming to "pinned" also...

Comment (gerhardol) [reply GitCommands/UserRepositoryHistory/Repository.cs]: Another massive change...
Rebased too.

File: GitUI/Translation/English.xlf [hunk 2]
```diff
@@ -6562,15 +6562,15 @@ Nothing to rebase.</source>
...
+        <source>Recent repositories</source>
         <target />
       </trans-unit>
       <trans-unit id="lblRecentRepositoriesHistorySize.Text">
         <source>Recent repositories history size</source>
         <target />
       </trans-unit>
       <trans-unit id="maxRecentRepositories.Text">
-        <source>Maximum number of most recent repositories</source>
+        <source>Maximum number of pinned recent repositories</source>
```
Comment (RussKie) [inline GitUI/Translation/English.xlf:6573]: Should be "Maximum number of pinned repositories"

Comment (gerhardol) [reply GitUI/Translation/English.xlf:6573]: This was intentional: The description is intended to differentiate pinned from favorites

Comment (RussKie) [reply GitUI/Translation/English.xlf:6573]: This made me question the difference between the pinned and favourites. I do feel that there's a significant overlap between these two, and a lot of added complexity. If we look at VS or other IDEs or productivity apps - most of those have a notion of "recent"s, and some have notion of "pinned" projects/files/etc.

Comment (gerhardol) [reply GitUI/Translation/English.xlf:6573]: I say the GE repo favorites is a good name, there are categories and such.
In general, Pinned is similar to Favorites but you only select a certain items, no categories.

Most recent is in my opinion a better name than Pinned, as this is not something you select yourself. As both you and @pmiossec suggested the name so I agreed, to have the change ready for the alpha.
Maybe *Top recent* is better name?
I can just revert the renaming to Pinned too if you prefer that.

**Comment by mstv** [comment]
Both the repo dropdown in the toolbar and the Dashboard are based on these lists.
This PR affects the repo dropdown in the way I want it: "less recent repositories" turned into "all recent repositories". 👍
But here "pinned" for "most recent" is / would be arguable. It still works as "most recent" and this is just fine.
(I don't need the Dashboard. I am used to the pinned Windows taskbar icon with its jump list and / or the repo dropdown if the jump list is incomplete.)

image

**Comment by gerhardol** [comment]
> But here "pinned" for "most recent" is / would be arguable. It still works as "most recent" and this is just fine.

Agree. I got two votes to change this and was at that time OK with it but should have argued...
But most recent requires you to think a little.
Personally, I do not feel it is worth it to change the naming.

**Pull request merged**